### ISSUE TITLE
Increase test coverage

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#672]
Current test coverage is 74% -- not bad!

Tasks:
- [x] **calibration**
    - [x] everything (run in debug mode?)
- [x] **sim**
    - [x] sim.shrink
    - [x] sim.save
    - [x] sim.export_pars
    - [x] sim.to_json
- [x] **multisim**
    - [x] multisim.show
    - [x] multisim.init_sims
    - [x] multisim.reset
    - [x] multisim.summarize
    - [x] run:single_run
    - [ ] run:multi_run
    - [x] run:parallel
- [x] **dist**
    - [x] dist.show_state
    - [x] dist.plot_hist
    - [ ] randint
    - [ ] multi_random:combine_rvs
    - [ ] trigger distseedrepeaterror
- [x] **networks**
    - [x] networks.validate
    - [x] network.get_inds
    - [x] network.pop_inds
    - [x] network.find_edges
    - [ ] randomnet.get_source
    - [x] msmnet
- [x] **loop**
    - [x] loop.run_one_step
    - [x] loop.to_df
    - [x] loop.plot
    - [x] loop.plot_cpu
- [x] **results**
    - [x] result.to_df
    - [x] result.plot
    - [x] results.to_df
    - [x] results.plot
- [ ] **arrays**
    - [ ] arrays.check_dtype
    - [ ] settings.context
    - [ ] settings.disp
    - [ ] settings.set
- [x] **utils**
    - [x] utils.check_requires

Old:
```
Name                            Stmts   Miss Branch BrPart  Cover
-----------------------------------------------------------------
starsim/__init__.py                30      0      2      1    97%
starsim/arrays.py                 325     66    162     34    76%
starsim/calibration.py            328    143    120     27    53%
starsim/demographics.py           309     14     54      6    94%
starsim/disease.py                204     17     76     10    90%
starsim/diseases/__init__.py        8      0      0      0   100%
starsim/diseases/cholera.py        86     10     10      1    86%
starsim/diseases/ebola.py          61      0      6      0   100%
starsim/diseases/gonorrhea.py      39      0      0      0   100%
starsim/diseases/hiv.py            92     39      8      0    55%
starsim/diseases/measles.py        44      0      6      0   100%
starsim/diseases/ncd.py            46      0      2      0   100%
starsim/diseases/sir.py           110     22      8      0    78%
starsim/diseases/syphilis.py      195     12     36      3    94%
starsim/distributions.py          637    118    216     39    80%
starsim/interventions.py          221     34     42     11    81%
starsim/loop.py                   202     85     56      2    60%
starsim/modules.py                205     34     76      9    81%
starsim/networks.py               495     96     98     13    77%
starsim/parameters.py             278     63    136     28    75%
starsim/people.py                 173     29     64     11    79%
starsim/products.py                85      8     26      4    86%
starsim/results.py                153     68     56      7    53%
starsim/run.py                    265    121    110     13    50%
starsim/samples.py                161     57     92      6    61%
starsim/settings.py               106     59     26      0    37%
starsim/sim.py                    447    124    200     31    66%
starsim/timepars.py               197     32     80     18    81%
starsim/utils.py                  193     58     92     20    66%
starsim/version.py                  4      0      0      0   100%
TOTAL                            5699   1309   1860    294    74%

```

**Comment by cliffckerr** [comment]
- [ ] **calibration
- [ ] everything (run in debug mode?)
- [ ] **sim**
- [ ] sim.shrink
- [ ] sim.save
- [ ] sim.export_pars
- [ ] sim.to_json
- [ ] **multisim**
- [ ] multisim.show
- [ ] multisim.init_sims
- [ ] multisim.reset
- [ ] multisim.summarize
- [ ] run:single_run
- [ ] run:multi_run
- [ ] run:parallel
- [ ] **dist**
- [ ] dist.show_state
- [ ] dist.plot_hist
- [ ] randint
- [ ] multi_random:combine_rvs
- [ ] trigger distseedrepeaterror
- [ ] networks.validate
- [ ] network.get_inds
- [ ] network.pop_inds
- [ ] network.find_edges
- [ ] randomnet.get_source
- [ ] msmnet
- [ ] **loop**
- [ ] loop.run_one_step
- [ ] loop.to_df
- [ ] loop.plot
- [ ] loop.plot_cpu
- [ ] **result**
- [ ] result.to_df
- [ ] result.plot
- [ ] **results**
- [ ] results.to_df
- [ ] results.plot
- [ ] **arrays**
- [ ] arrays.check_dtype
- [ ] settings.context
- [ ] settings.disp
- [ ] settings.set
- [ ] **utils**
- [ ] utils.check_requires

**Comment by cliffckerr** [comment]
Test coverage now 81%, including the important functions. Closed by #673
